Readme.md

```diff
@@ -386,6 +388,174 @@ export function MyForm({ isNewItem, addItem, updateItem }: Props): JSX.Element =
 
 ---
 
+## Nested Objects
+
+Each FormField object is also able to contain fields. It behaves similar as with the Form object but with slight differences on it's methods (See [API](#nested-fields)).
+
+```ts
+import { useForm } from 'react-use-form-library';
+
+function App(): JSX.Element {
+  const { model, fields, onSubmit } = useForm({
+    model: {
+      address: {
+        streetName: '',
+      },
+    },
+    handleSubmit: async () => {
+      console.log(model);
+    },
+  });
+
+  return (
+    <form onSubmit={onSubmit}>
+      <input
+        /* Since the FormField could be either a value or an object, the fields property is optional 
+        and you will need to check for existance */
+        value={fields.address.fields?.streetName.value}
+        onChange={(v) =>
+          fields.address.fields?.streetName.onChange(v.target.value)
+        }
+      />
+      <button>Submit</button>
+    </form>
+  );
+}
+```
+
+Validations also work with nested objects
+
+```ts
+import { useForm } from 'react-use-form-library';
+
+function App(): JSX.Element {
+  const { model, fields, onSubmit } = useForm({
+    model: {
+      address: {
+        streetName: '',
+      },
+    },
+    handleSubmit: async () => {
+      console.log(model);
+    },
+    validations: {
+      address: {
+        streetName: 'required',
+      },
+    }
+  });
+
+  // ...
+```
+
+Since the fields property within FormField is also based on a proxy object, you can access nested properties without
+explicitely declaring them
+
+```ts
+function App(): JSX.Element {
+  const { model, fields, onSubmit } = useForm({
+    model: {},
+    handleSubmit: async () => {
+      console.log(model);
+    },
+  });
+
+  return (
+    <form onSubmit={onSubmit}>
+      <input
+        value={fields.bankDetails.fields?.address.fields?.streetName.value}
+        onChange={(v) =>
+          fields.bankDetails.fields?.address.fields?.streetName.onChange(
+            v.target.value
+          )
+        }
+      />
+      <button>Submit</button>
+    </form>
+  );
+}
+```
+
+---
+
+## FieldSet
+
+A FieldSet is basically an array of FormFields. Contrary to nested objects, a FieldSet will only be instantiated if it is **explicitely** declared in the model.
+
+This is because at the moment of field instantiation, the addField method within a Form (or a FormField if it is a nested object) will as if the model value is actually an Array. It is not possible of implicitely determine if the field is an array based only on the type.
```

Comment:
```suggestion
This is because at the moment of field instantiation, the `addField` method within a `Form` (or a `FormField` if it is a nested object) will as if the model value is actually an Array. It is not possible to implicitly determine if the field is array-based only on the type.
```